### PR TITLE
stop service at ceiling height

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: committee/committee.go [hunk 1]
```diff
@@ -227,7 +230,8 @@ func (ec *committee) Start(ctx context.Context) (err error) {
 		for {
 			select {
 			case <-ec.terminate:
-				ec.terminate <- true
+				ec.carrier.Close()
+				err = ec.archive.Stop(ctx)
```
Comment (dustinxie) [inline committee/committee.go:234]: don't know why do "ec.terminate <- true"
shouldn't we stop here?

File: committee/committee.go [hunk 2]
```diff
@@ -227,13 +230,15 @@ func (ec *committee) Start(ctx context.Context) (err error) {
 		for {
 			select {
 			case <-ec.terminate:
-				ec.terminate <- true
```
Comment (dustinxie) [inline committee/committee.go:230]: don't understand this

in current code, ec.terminate is unbuffered channel, the goroutine will block here and won't return

Comment (yutongp) [inline committee/committee.go:230]: this line is intended  to stop another go routine. subscribeNewblock

File: committee/committee.go [hunk 3]
```diff
@@ -227,13 +230,15 @@ func (ec *committee) Start(ctx context.Context) (err error) {
...
 				return
 			case tip := <-tipChan:
 				zap.L().Info("new ethereum block", zap.Uint64("height", tip))
 				if err := ec.Sync(tip); err != nil {
 					zap.L().Error("failed to sync", zap.Error(err))
 				}
+				if ec.currentHeight >= ec.ceilingHeight {
+					ec.terminate <- true
```
Comment (yutongp) [inline committee/committee.go:240]: return here

File: committee/committee.go (hunk 2 above)
Comment (dustinxie) [reply committee/committee.go:230]: thx, improved in latest push

File: committee/committee.go (hunk 3 above)
Comment (dustinxie) [reply committee/committee.go]: fixed in latest push

**Comment by dustinxie** [comment]
used #146

**Pull request closed without merging**